app/api/routers/breeze_buddy/leads/rbac.py

```
@@ -0,0 +1,101 @@
+    if current_user.role == "admin":
+        return
+
+    # Check merchant access
+    if lead.merchant_id not in current_user.merchant_ids and "*" not in current_user.merchant_ids:
+        logger.warning(
+            f"User {current_user.username} attempted to {operation} lead {lead.id} "
+            f"for unauthorized merchant: {lead.merchant_id}"
+        )
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"Lead not found"
```

Ruff (0.14.8)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)